src/bazel/webpack.bzl

```diff
@@ -0,0 +1,62 @@
+def _collect_es5_sources_impl(target, ctx):
+  result = set()
+  if hasattr(ctx.rule.attr, "srcs"):
+    for dep in ctx.rule.attr.srcs:
+      if hasattr(dep, "es5_sources"):
+        result += dep.es5_sources
+  if hasattr(target, "typescript"):
+    result += target.typescript.es5_sources
+  return struct(es5_sources = result)
+
+_collect_es5_sources = aspect(
+    _collect_es5_sources_impl,
+    attr_aspects = ["deps", "srcs"],
+)
+
+def _webpack_bundle_impl(ctx):
+  inputs = set()
+  for s in ctx.attr.srcs:
+    if hasattr(s, "es5_sources"):
+      inputs += s.es5_sources
+
+  config = ctx.attr.config.files.to_list()[0]
+
+  if ctx.attr.mode == 'prod':
+    main = ctx.new_file('bundles/main.bundle.prod.js')
+    polyfills = ctx.new_file('bundles/polyfills.bundle.prod.js')
+    vendor = ctx.new_file('bundles/vendor.bundle.prod.js')
+    styles = ctx.new_file('bundles/styles.bundle.prod.js')
+  else:
+    main = ctx.new_file('bundles/main.bundle.js')
+    polyfills = ctx.new_file('bundles/polyfills.bundle.js')
+    vendor = ctx.new_file('bundles/vendor.bundle.js')
+    styles = ctx.new_file('bundles/styles.bundle.js')
+
+  inputs += [config]
+  args = []
+
+  if ctx.attr.mode == 'prod':
+    args += ['-p']
+
+  args += ['--config', config.path]
+  args += ['--env.bin', ctx.configuration.bin_dir.path]
```

Comment:
nit: `bin_dir` is consistent and won't be confused as an argument providing a binary to execute